### PR TITLE
BAH-2179 | To integrate sms in registration

### PR DESCRIPTION
**Pull request by SanoferSameera**

### CODE REVIEW

File: ui/app/registration/constants.js [hunk 1]
```diff
@@ -19,10 +19,12 @@ Bahmni.Registration.Constants = {
     allAddressFileds: ["uuid", "preferred", "address1", "address2", "address3", "address4", "address5", "address6", "cityVillage", "countyDistrict", "stateProvince", "postalCode", "country", "latitude", "longitude"],
     nextStepConfigId: "org.bahmni.registration.patient.next",
     patientNameDisplayOrder: ["firstName", "middleName", "lastName"],
-    whatsAppMessage: "REGISTRATION_WHATSAPP_MESSAGE",
+    registrationMessage: "REGISTRATION_MESSAGE",
     enableWhatsAppButton: false,
     enableDashboardRedirect: false,
-    dashboardUrl: "/bahmni/clinical/index.html#/default/patient/{{patientUuid}}/dashboard"
+    dashboardUrl: "/bahmni/clinical/index.html#/default/patient/{{patientUuid}}/dashboard",
+    otpServiceUrl: hostUrl + "/otp-service",
```
Comment (gsluthra) [inline ui/app/registration/constants.js:26]: is there an assumption that otpservice has to be in same location as host url?

File: .github/workflows/build_publish.yml [hunk 2]
```diff
@@ -4,6 +4,7 @@ on:
     branches:
       - master
       - 'release-*'
+      - BAH-2179
```
Comment (swatigogia2020) [inline .github/workflows/build_publish.yml:7]: This needs to be removed

File: ui/app/registration/controllers/editPatientController.js [hunk 3]
```diff
@@ -98,11 +98,8 @@ angular.module('bahmni.registration')
             };
 
             $scope.sendWhatsAppMessage = function () {
```
Comment (swatigogia2020) [inline ui/app/registration/controllers/editPatientController.js:100]: can we re-name this function

File: ui/app/registration/services/defaultPatientServiceStrategy.js [hunk 4]
```diff
@@ -4,7 +4,6 @@ angular.module('bahmni.registration')
     .service('patientServiceStrategy', ['$http', '$q', '$rootScope', function ($http, $q, $rootScope) {
         var openmrsUrl = Bahmni.Registration.Constants.openmrsUrl;
         var baseOpenMRSRESTURL = Bahmni.Registration.Constants.baseOpenMRSRESTURL;
-
         var search = function (config) {
```
Comment (swatigogia2020) [inline ui/app/registration/services/defaultPatientServiceStrategy.js:7]: can we revert the change in this file.

**Review by swatigogia2020 — CHANGES_REQUESTED** [review]

File: .github/workflows/build_publish.yml (hunk 2 above)
Comment (kavitha-sundararajan) [reply .github/workflows/build_publish.yml]: removed the branch

File: ui/app/registration/controllers/editPatientController.js (hunk 3 above)
Comment (kavitha-sundararajan) [reply ui/app/registration/controllers/editPatientController.js]: renamed the function to notifyOnWhatsApp

**Review by gsluthra — CHANGES_REQUESTED** [review]

File: ui/app/registration/services/patientService.js [hunk 5]
```diff
@@ -90,6 +91,28 @@ angular.module('bahmni.registration')
...
+            var message = $translate.instant(appService.getAppDescriptor().getConfigValue("registrationMessage") || Bahmni.Registration.Constants.registrationMessage);
+            message = message.replace("#patientId", patientId);
+            message = message.replace("#name", name);
+            message = message.replace("#age", age + " years");
+            message = message.replace("#gender", gender);
+            return message;
+        };
+
+        const smsAlert = function (patientId, name, age, gender, phoneNumber) {
+            const url = otpServiceUrl + "/notification/sms";
```
[the remark this answers is not in the document]
Comment (kavitha-sundararajan) [reply ui/app/registration/services/patientService.js]: Segregated the sms functionality to a separate service

**Review by swatigogia2020 — APPROVED** [review]

File: ui/app/registration/services/patientService.js [hunk 6]
```diff
@@ -90,6 +90,18 @@ angular.module('bahmni.registration')
             return $http.post(url, data, config);
         };
 
+        var getRegistrationMessage = function (patientId, name, age, gender) {
+            var clinicName = $rootScope.loggedInLocation.name;
+            var message = $translate.instant(appService.getAppDescriptor().getConfigValue("registrationMessage") || Bahmni.Registration.Constants.registrationMessage);
+            message = message.replace("#clinicName", clinicName);
+            message = message.replace("#patientId", patientId);
+            message = message.replace("#name", name);
+            message = message.replace("#age", age + " years");
```
Comment (gsluthra) [inline ui/app/registration/services/patientService.js:99]: "years" should also be in transaltion message

File: ui/app/registration/services/patientService.js [hunk 7]
```diff
@@ -96,7 +96,7 @@ angular.module('bahmni.registration')
             message = message.replace("#clinicName", clinicName);
             message = message.replace("#patientId", patientId);
             message = message.replace("#name", name);
-            message = message.replace("#age", age + " years");
+            message = message.replace("#age", age + " " + $translate.instant("REGISTRATION_LABEL_YEARS"));
```
[the remark this answers is not in the document]
Comment (kavitha-sundararajan) [reply ui/app/registration/services/patientService.js]: refactored to read it from the message itself
